src/dimension_slice.c

```diff
@@ -101,6 +145,8 @@ dimension_slice_scan_limit(int32 dimension_id, int64 coordinate, int limit)
 	ScanKeyData scankey[3];
 	DimensionVec *slices = dimension_vec_create(limit > 0 ? limit : DIMENSION_VEC_DEFAULT_SIZE);
 
+	coordinate = GET_COORDINATE_FOR_SLICE(coordinate);
```

Comment:
I am indifferent about whether this should be a macro or not. I made it a macro just because the same logic is used in several places.  I can move this logic into the code directly if you think that would be easier to understand.

Reply:
I do think that the dimension_slice module should be the only code that knows about this logic. I don't want code in other modules (i.e. `hyperspace_calculate_point()`) knowing about this logic. It's internal implementation details of dimension_slice. Notice that even the constant definitions for max and min values are defined in the .c file not the .h. I want to hide this stuff from the rest of the system.